File: php/test/Structural/AdapterTest.php
<?

require_once 'src/Structural/Adapter/TeaCup.php';
require_once 'src/Structural/Adapter/TeaBag.php';
require_once 'src/Structural/Adapter/TeaBall.php';
require_once 'src/Structural/Adapter/LooseLeafTea.php';

class Structural_AdapterTest extends PHPUnit_Framework_TestCase
{
   public function test()
   {
		$teaCup = new TeaCup();
		$teaBag = new TeaBag();       
		$teaCup->steepTeaBag($teaBag);
		$this->assertEquals($teaBag->teaBagIsSteeped, 1, 'Test that tea bag cup steeped tea');

		$looseLeafTea = new LooseLeafTea();
		$teaBall = new TeaBall($looseLeafTea);
		$teaCup->steepTeaBag($teaBall);
		$this->assertEquals($teaBall->teaBagIsSteeped, 1, 'Test that tea ball steeped teabag');

	}
}

File: clojure/test/patterns/structural/adapter_test.clj
(ns patterns.structural.adapter-test
	(:require
      [clojure.test :refer :all]
      [patterns.structural.adapter :refer :all])
   (:import (patterns.structural.adapter teabag teaball looseleaftea)))

(deftest structural-adapter-test
	(let [
		 a-teabag (make-teabag)
		 some-looseleaf (make-looseleaftea)
		 a-teaball (make-teaball some-looseleaf) ]

	(testing "Can steep-tea-bag with a teabag" 
		(is (= true (:teabag-is-steeped (steep-tea-bag a-teabag)))))

	(testing "Can steep-tea-bag with a teaball" 
		(is (= true (:teabag-is-steeped (steep-tea-bag a-teaball)))))

	))

